module/users/helper.js

```diff
@@ -498,34 +498,63 @@ module.exports = class UsersHelper {
-          importedSurveys = importedSurveys.result
-          let importedObservations = await surveyService.userObservations(token,programId)
-          importedObservations = importedObservations.result
-          let importedProjects = await improvementProjectService.importedProjects(token,programId); 
-          
-          if(importedProjects.data.length> 0 && (type ==="" || type === constants.common.IMPROVEMENT_PROJECT)){
-            importedProjects.data.forEach((importedProject) => {
-              solutionIds.push(importedProject.solutionInformation._id)
-            })
-          }
-          
-          if(importedSurveys.length> 0 && (type ==="" || type === constants.common.SURVEY)){
-            importedSurveys.forEach((surveys)=>{
-              solutionIds.push(surveys.solutionId)
-            })
-          }
-          if(importedObservations.length> 0 && (type ==="" || type === constants.common.OBSERVATION)){
-            importedObservations.forEach((observation)=>{
-              solutionIds.push(observation.solutionId)
-            })
+          const solutionIds = []
+          const getAllResources = []
+
+          /**
+           * @function importedProjects
+           * @function userSurveys
+           * @function userObservations
+           * 
+           * @param token string: userToken
+           * @param programId string: programId
+           * 
+           * @returns {Promise} 
+           * 
+           */
+          // Creates an array of promises based on users Input
+          switch(type){
+            case constants.common.IMPROVEMENT_PROJECT:
+              getAllResources.push(improvementProjectService.importedProjects(token,programId));
+              break;
+            case constants.common.SURVEY:
+              getAllResources.push(surveyService.userSurveys(token,programId));
+              break;
+            case constants.common.OBSERVATION: 
+              getAllResources.push(surveyService.userObservations(token,programId));
+              break;
+            default:
+              getAllResources.push(improvementProjectService.importedProjects(token,programId));
+              getAllResources.push(surveyService.userSurveys(token,programId));
+              getAllResources.push(surveyService.userObservations(token,programId));
           }
+          //here will wait till all promises are resolved
+          const allResources = await Promise.all(getAllResources)
+
+          //Will find all solutionId from response
+          allResources.forEach(resources => {
+            // this condition is required because it returns response in different object structure
+            if(resources.type === constants.common.IMPROVEMENT_PROJECT && resources.success === true){
+              resources.data.forEach(importedProject => {
+                solutionIds.push(importedProject.solutionInformation._id)
+              });
+            }else if(resources.success === true) {
```

Comment:
@ankitpws Please format and remove extra lines

Reply:
Done